javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+                logger.error("S3Exception occurred: {}", cause.getMessage(), ce);
+            } else {
+                logger.error("An unexpected error occurred: {}", cause.getMessage(), ce);
+            }
+        }
+        logger.info(""" 
+            It worked! It's important to remember the user permissions when interacting with 
+            Directory buckets. Instead of validating permissions on every call as 
+            normal buckets do, Directory buckets utilize the user credentials and session 
+            token to validate. This allows for much faster connection speeds on every call. 
+            For single calls, this is low, but for many concurrent calls 
+            this adds up to a lot of time saved.
+            """);
+        waitForInputToContinue(scanner);
+        return bucketObject;
+    }
+
+    /**
+     * Creates VPC users for the S3 Express One Zone scenario.
+     * <p>
+     * This method performs the following steps:
+     * <ol>
+     *     <li>Optionally creates a new VPC and VPC Endpoint if the application is running in an EC2 instance in the same Availability Zone as the Directory buckets.</li>
+     *     <li>Creates two IAM users: one with S3 Express One Zone permissions and one without.</li>
+     * </ol>
+     *
+     * @return a {@link UserNames} object containing the names of the created IAM users
+     */
+    public static UserNames createVpcUsers() {
+        /*
+        Optionally create a VPC.
+        Create two IAM users, one with S3 Express One Zone permissions and one without.
+        */
+        logger.info("""
+            1. First, we'll set up a new VPC and VPC Endpoint if this program is running in an EC2 instance in the same AZ as your\s
+            Directory buckets will be. Are you running this in an EC2 instance located in the same AZ as your intended Directory buckets?
+            """);
+
+        logger.info("Do you want to setup a VPC Endpoint? (y/n)");
+        String endpointAns = scanner.nextLine().trim();
+        if (endpointAns.equalsIgnoreCase("y")) {
+            logger.info("""
+                Great! Let's set up a VPC, retrieve the Route Table from it, and create a VPC Endpoint to connect the S3 Client to.
+                """);
+            try {
+                s3DirectoriesActions.setupVPCAsync().join();
+            } catch (CompletionException ce) {
+                Throwable cause = ce.getCause();
+                if (cause instanceof Ec2Exception) {
+                    logger.error("IamException occurred: {}", cause.getMessage(), ce);
+                } else {
+                    logger.error("An unexpected error occurred: {}", cause.getMessage(), ce);
+                }
+            }
+            waitForInputToContinue(scanner);
+        } else {
+            logger.info("Skipping the VPC setup. Don't forget to use this in production!");
+        }
+        logger.info("""            
+            2. Policies, users, and roles with CDK.
```

Comment:
I suggest that you create a separate method for the CloudFormation Stack creation and call it from the Scenario. It's buried here, but should be at the same level as other calls from the Scenario file.